systemtest/benchtest/main_test.go

```diff
@@ -37,166 +37,56 @@ import (
 
 func Test_warmup(t *testing.T) {
 	type testCase struct {
-		agents int
-		events []uint
+		agents   int
+		duration time.Duration
 	}
 	cases := []testCase{
-		{1, []uint{100, 1000}},
+		{1, time.Second},
+		{10, time.Second},
 	}
 	for _, c := range cases {
-		for _, events := range c.events {
-			t.Run(fmt.Sprintf("%d_agent_%v_events", c.agents, events), func(t *testing.T) {
-				var received uint64
-				srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-					if r.URL.Path == "/debug/vars" {
-						// Report idle APM Server.
-						w.Write([]byte(`{"libbeat.output.events.active":0}`))
-					}
-
-					if !strings.HasPrefix(r.URL.Path, "/intake") {
+		t.Run(fmt.Sprintf("%d_agent_%s_events", c.agents, c.duration.String()), func(t *testing.T) {
+			var received uint64
+			srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				if r.URL.Path == "/debug/vars" {
+					// Report idle APM Server.
+					w.Write([]byte(`{"libbeat.output.events.active":0}`))
+				}
```

Comment:
**issue**: we should return early here. if `w.Write` is called the server has already sent a response to the client